tests/crypto/test_cases/test_vectors_aead_ccm.c

```diff
@@ -0,0 +1,4476 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <string.h>
+#include <stddef.h>
+
+#include "common_test.h"
+#include <mbedtls/ccm.h>
+
+/**@brief AES CCM test vectors can be found on NIST web pages.
+ *
+ * https://csrc.nist.gov
+ * https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Algorithm-Validation-Program/documents/mac/ccmtestvectors.zip
+ */
+
+#if defined(MBEDTLS_AES_C)
```

Comment:
Adding a source file to the build and then ifdef'ing the whole thing out is bad practice. We should instead conditionally add the source file to the build depending on if a Kconfig option is enabled. I assume there exists a Kconfig option called MBEDTLS_AES_C or similar.

Reply:
Changed in adc45bd , both test files and test vector sources are now added conditionally.
There is still some granularity in test vector sources based on some more specific configurations, but there are now no `#if defined(...)` blocks encompassing entire files.

Reply:
Very nice.